### ISSUE TITLE
Network VBN API Authorization

### ISSUE THREAD

**Issue by tinxx** [grote/Transportr#130]
Getting the above exception when trying to navigate using VBN.

**Comment by grote** [comment]
Unfortunately, the VBN now requires Authorization to use their API.

   https://github.com/grote/Transportr/blob/master/src/de/grobox/liberario/NetworkProviderFactory.java#L306

Would you like to help to search for information on their website about this and if you can't find anything contact them to get this authorization for Transportr?

**Comment by grote** [comment]
I haven't heard back from you, so I am going to close this issue. Let me know when I should reopen this.

**Comment by tinxx** [comment]
They haven't yet decided if they are going to license the necessary API.
I hope to hear from them soon and will report back when that is the case.

Should I file another ticket on that  issue?

**Comment by grote** [comment]
No, we can keep using this one. I will reopen it. Didn't know that you are actually in contact with them. That's good news. Please keep us updated.

Did why say why it is no problem to give access to Oeffi, but to Transportr?

**Comment by tinxx** [comment]
Actually they were perplexed when I told them that it worked with this app in the past.
I didn't ask them about Öffi.

**Comment by grote** [comment]
Any news here?

**Comment by tinxx** [comment]
Sadly, no. No news, yet.

**Comment by andreas-h** [comment]
Since this doesn't seem to work (actually, I don't see VBN listed as Verkehrsverbund option), maybe it'd be good to remove Bremen from the map of supported places?

**Comment by grote** [comment]
> maybe it'd be good to remove Bremen from the map of supported places?

Done, thanks!

**Comment by grote** [comment]
Since it is free software now, we could maybe use those credentials: https://gitlab.com/oeffi/oeffi/blob/350f686e667c4557c2450d0ac5e828eaddc385a9/oeffi/src/de/schildbach/oeffi/network/NetworkProviderFactory.java#L123

Pull requests welcome! It would need to be added here: https://github.com/grote/Transportr/blob/master/app/src/main/java/de/grobox/transportr/networks/TransportNetworks.kt#L53

**Comment by katehowler** [comment]
@tinxx, can you please post your correspondence with the [VBN](https://www.vbn.de/) here?

**Comment by tinxx** [comment]
@katehowler, I only found this one in my inbox.
TLDR: Back in 2015 they only offered arrival / departure information for individual stations including forecasts but no routing information.

> From: VBN-FahrPlaner Support <App@vbn.de>
> Subject: AW: Zugang zu Verbindungsdaten
> Date: 2015-12-16, 13:31
>
> Hallo Herr ***,
> 
> vielen Dank für Ihre Anfrage! Wir bieten unseren Kunden mit Android-Smartphones die App auch als direkt zu installierende apk-Datei an, so dass sie nicht gezwungen sind GooglePlay zu nutzen. Bei iOS ist dieses Verfahren leider nicht möglich. 
> 
> Sie schreiben, dass die Abfrage von Verbindungsauskünften per API nicht *mehr* möglich sei. Sind Sie sich sicher, dass Sie "fahrplaner.vbn.de" meinen? Wir haben eine solche API über unser Systemhaus nämlich bisher noch gar nicht lizensieren lassen, da es sich um relativ hohe Invest- und Betriebskosten handelt. 
> 
> Allerdings bieten wir Interessenten die Nutzung einer entsprechenden API zum Abruf von Stationsbezogenen Ankunft-/Abfahrtstafeln inkl. Echtzeitprognosen an. Wäre Ihnen damit evtl. weiter geholfen?
> 
> 
> Mit freundlichen Grüßen
> 
> i. A. ***
> Echtzeitinformationssysteme
> 
> Verkehrsverbund Bremen/Niedersachsen GmbH (VBN)
> Willy-Brandt-Platz 7
> 28215 Bremen
> www.vbn.de
